db/compaction/compaction.cc

```diff
@@ -207,11 +207,12 @@ Compaction::Compaction(VersionStorageInfo* vstorage,
                        const MutableCFOptions& _mutable_cf_options,
                        std::vector<CompactionInputFiles> _inputs,
                        int _output_level, uint64_t _target_file_size,
-                       uint64_t _max_compaction_bytes, uint32_t _output_path_id,
+                       uint64_t _max_compaction_bytes,
                        CompressionType _compression,
                        CompressionOptions _compression_opts,
                        uint32_t _max_subcompactions,
                        std::vector<FileMetaData*> _grandparents,
+                       DbPathSupplier* db_path_supplier,
```

Comment:
It looks like that we are passing ownership to Compaction class. If that is the case, the best practice is to use `std::unique_ptr<DbPathSupplier>&&`.

Reply:
Changed to std::unique_ptr<DbPathSupplier>&& for both compaction and flush job.